testutils/src/main/java/com/microsoft/identity/internal/testutils/CacheUtils.java

```diff
@@ -0,0 +1,147 @@
+     * @param predicate      Generic functional interface representing function that returns true
+     *                       or false depending on taken type.
+     * @param editor         Functional interface to have any number of token editing method.
+     */
+    public void editAllTokenInCache(@NonNull final String sharedPrefName, Predicate<String> predicate, Function<String, String> editor) {
+        SharedPreferences sharedPref = TestUtils.getSharedPreferences(sharedPrefName);
+        SharedPreferences.Editor prefEditor = sharedPref.edit();
+        Map<String, ?> cacheEntries = sharedPref.getAll();
+
+        //get all the key from the cache entry, verify and edit it.
+        for (Map.Entry<String, ?> cacheEntry : cacheEntries.entrySet()) {
+            String keyToEdit = cacheEntry.getKey();
+            if (predicate.test(keyToEdit)) {
+                Object cacheValue = cacheEntries.get(keyToEdit);
+                if (cacheValue instanceof String) {
+                    Credential credential = CACHE_KEY_VALUE_DELEGATE.fromCacheValue((String) cacheValue, Credential.class);
+                    credential.setSecret(editor.apply(credential.getSecret()));
+                    prefEditor.putString(keyToEdit, CACHE_KEY_VALUE_DELEGATE.generateCacheValue(credential));
+                    prefEditor.apply();
+                }
+            }
+        }
+    }
+
+    /**
+     * utility function to edit Access Token.
+     *
+     * @param sharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllAccessTokenInCache(@NonNull final String sharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return TestUtils.isAccessToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(sharedPrefName, predicate, editor);
+    }
+
+    /**
+     * utility function to edit Refresh Token.
+     *
+     * @param sharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllRefreshTokenInCache(@NonNull final String sharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return TestUtils.isRefreshToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(sharedPrefName, predicate, editor);
+    }
+
+    /**
+     * Split the string into 3 segment separated by period "." and set the
+     * random character at the random position in the 3rd segment.
+     *
+     * @param string string to be edited.
+     * @return edited string.
+     */
+    private static String editTokenSignature(String string) {
```

Comment:
nit: maybe randomizeCharacterInTokenSignature

Reply:
change the method name.